src/nvim/option.c

```diff
@@ -1511,9 +1511,10 @@ do_set (
           } else if (opt_idx >= 0) {                      /* string */
             char_u      *save_arg = NULL;
             char_u      *s = NULL;
-            char_u      *oldval;                /* previous value if *varp */
+            char_u      *oldval = NULL;         /* previous value if *varp */
```

Comment:
`//` comment. Since you are already changing the line, it is better to also change the comment.